sigridci/sigridci.py

```diff
@@ -156,14 +154,24 @@ def getRequestUploadPath(self, systemExists):
         return path
         
     def uploadBinaryFile(self, url, upload):
+        for attempt in range(self.RETRY_ATTEMPTS):
+            try:
+                return self.attemptUpload(url, upload)
+            except urllib.error.HTTPError as e:
+                log("Retrying upload")
+                time.sleep(self.POLL_INTERVAL)
+        
+        log("Sigrid is currently unavailable")
+        sys.exit(1)
+        
+    def attemptUpload(self, url, upload):
         with open(upload, "rb") as uploadRef:
-            uploadRequest = urllib.request.Request(url, data=uploadRef.read())
+            uploadRequest = urllib.request.Request(url, data=uploadRef)
             uploadRequest.method = "PUT"
             uploadRequest.add_header("Content-Type", "application/zip")
             uploadRequest.add_header("Content-Length", "%d" % os.path.getsize(upload))
             uploadRequest.add_header("x-amz-server-side-encryption", "AES256")
-            uploadResponse = urllib.request.urlopen(uploadRequest)
-            return uploadResponse.status in [200, 201, 202]
+            return urllib.request.urlopen(uploadRequest)
```

Comment:
There's no more need to return a result here and subsequently in `uploadBinaryFile` anymore right, as we no longer check the status, but `sys.exit(1)` instead when we run out of attempts right?

Reply:
Indeed, we could drop the return value.